### PR TITLE
Issue #1 | Connect app to Firebase

### LINKED ISSUE #1: 1. As a component, I want to be able to read from and write to the Firestore database so that users can persist information.
This is 1 of 2 stories that should be completed in the first week to enable further development of the app.

AC:

* The following have been added as project dependencies: `firebase` & `react-firestore`
* Make a change in the Firestore database and it shows up in the app
* Make a change in the app and it shows up in the Firestore database

### PR DESCRIPTION
**Pull request by daredtech**
## Description

Added three new components: one for referencing the database, one for adding an item to the database, and another one to display items from the database in real-time.

### Referencing database:
Inside Firebase component initialized the dataBase that would allow accessing and updating the firestore.

### Displaying items from the database:
Inside Listener component set up a listener that would monitor the updates in the database and display the changes in real-time. 

### Writing items to the database:
Inside AddItem component wrote a function that would update the database every time a new item is added

## Related Issue

Closes https://github.com/the-collab-lab/tcl-11-smart-shopping-list/issues/1

## Acceptance Criteria

- [x] The following have been added as project dependencies: firebase & react-firestore
- [X] Make a change in the Firestore database and it shows up in the app
- [X] Make a change in the app and it shows up in the Firestore database

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria

From the terminal, pull down this branch with `git pull origin fc-vh-connecting-to-firestore` and check that branch out with `git checkout c-vh-connecting-to-firestore`
Run `npm install` to install the newly added dependencies locally. Run `npm start` to launch the app.
Click on `Click to get items!` button to display items in the database.
Type the name of the item you'd like to add and click `Add Item` button to add the item to the database. The newly added item should be displayed now.

### CODE REVIEW

**Review by CodeMeKathy — APPROVED** [review]
### All Issue: #1 Acceptance Criteria (AC) successfully met.

#### Recommendations/Suggestions: 
- Add parent directory to house `<AddItem />` component
- Review and please consider incorporating the suggested file structure and file naming convention on `bl-kl-create-view-navigation` branch or provide your own suggestion to the `fc-vh-connecting-to-firestore` branch

**Comment by daredtech** [comment]
@CodeMeKathy @brewswain Thank you for your feedback. I updated the structure and changed the file names.

File: src/App.js [hunk 1]
```diff
@@ -1,11 +1,20 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { useState } from 'react';
+import AddItem from './AddItem';
+import Listener from './Listener';
 import './App.css';
+// import logo from './logo.svg';
```
Comment (joeylaguna) [inline src/App.js]: This can be removed

File: src/App.js [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
+  const [itemsDisplay, setItemsDisplay] = useState(false);
+
   return (
     <div className="App">
-      <header className="App-header">
+      <button onClick={() => setItemsDisplay(true)}>Click to get items!</button>
+      {itemsDisplay && <Listener />}
+      <AddItem />
+
+      {/* <header className="App-header">
```
Comment (joeylaguna) [inline src/App.js]: This can be removed

**Review by joeylaguna — APPROVED** [review]

**Comment by joeylaguna** [comment]
Approved! Great job, team!

**Comment by joeylaguna** [comment]
Looks good! ✔️

**Pull request merged**